WindowsCompatibility/WindowsCompatibility.psm1

```diff
@@ -695,10 +695,18 @@ function Add-WindowsPSModulePath
         return
     }
 
-    $WindowsPSModulePath = [System.Environment]::GetEnvironmentVariable("psmodulepath", [System.EnvironmentVariableTarget]::Machine)
-    if (-not ($env:PSModulePath).Contains($WindowsPSModulePath))
+    $Paths = @(
+        "${env:userprofile}\Documents\WindowsPowerShell\Modules"
+        "${env:ProgramFiles}\WindowsPowerShell\Modules"
+        "${env:windir}\system32\WindowsPowerShell\v1.0\Modules"
+        [System.Environment]::GetEnvironmentVariable("PSModulePath", [System.EnvironmentVariableTarget]::Machine) -split [System.IO.Path]::PathSeparator
+    )
+
+    foreach ($Path in $Paths)
     {
-        $env:PSModulePath += ";${env:userprofile}\Documents\WindowsPowerShell\Modules;${env:programfiles}\WindowsPowerShell\Modules;${WindowsPSModulePath}"
+        if (-not $env:PSModulePath.Contains($Path))
+        {
+            $env:PSModulePath += ";$Path"
+        }
```

Comment:
I'd prefer something more exact when checking for duplicates. The simple string match does allow for prefix conditions. I would suggest using ordered hashtables to guarantee uniqueness. This might look something like:
```powershell

$paths =  @(
    $ENV:PSModulePath -split [System.IO.Path]::PathSeparator
    "${env:userprofile}\Documents\WindowsPowerShell\Modules"
    "${env:ProgramFiles}\WindowsPowerShell\Modules"
    "${env:windir}\system32\WindowsPowerShell\v1.0\Modules"
     [System.Environment]::GetEnvironmentVariable('PSModulePath',
         [System.EnvironmentVariableTarget]::Machine) -split [System.IO.Path]::PathSeparator
)

$pathTable = [ordered] @{}
foreach ($path in $paths)
{
    if ($pathTable[$path])
    {
        continue
    }
    $pathTable[$path] = $true
}
$ENV:PSModulePath = $pathTable.Keys -join [System.IO.Path]::PathSeparator
```

Reply:
I don't see why it is insufficient. Can you please expand on these prefix conditions?

Reply:
@sethvs lets say the first path in PSModulePath is `C:\foo\bar`. It gets added. Then the second path is `c:\foo`. Since `c:\foo\bar` contains `c:\foo`, `c:\foo` is detected as present by `$env:PSModulePath.Contains($Path)` and not added.

@BrucePay's suggestion of using the ordered hash makes for a fast lookup of the paths while also preserving order in away that wont exclude parent paths added after child paths.

Reply:
Thanks for the explanation.
Done.

Reply:
If you have `/a/b/c/d' in your path then try to add `/a/b/c, it won't work because the prefix '/a/b/c' is all ready there.